cmd/bundle/generate/utils.go

```diff
@@ -119,13 +146,15 @@ func (n *downloader) markNotebookForDownload(ctx context.Context, notebookPath *
 }
 
 func (n *downloader) FlushToDisk(ctx context.Context, force bool) error {
-	err := os.MkdirAll(n.sourceDir, 0o755)
-	if err != nil {
-		return err
-	}
-
 	// First check that all files can be written
 	for targetPath := range n.files {
+		// Create parent directories if they don't exist
+		dir := filepath.Dir(targetPath)
```

Comment:
Move directory creation below the validation check below? It's better if the and does not have side effects if we encounter an error.

Reply:
What could be side effects in this case?

Reply:
The directory would be created abd then the CLI would error. Ideally we would error and then create the directories.

Reply:
Why would `os.Stat` error in such cases? This is the only validation we do

Reply:
Yes, I think it makes sense indeed not to create directories in such cases, will update